Source/JavaScriptCore/runtime/ObjectConstructor.cpp

```diff
@@ -324,10 +324,49 @@ JSC_DEFINE_HOST_FUNCTION(objectConstructorAssign, (JSGlobalObject* globalObject,
     // FIXME: Extend this for non JSFinalObject. For example, we would like to use this fast path for function objects too.
     // https://bugs.webkit.org/show_bug.cgi?id=185358
     bool targetCanPerformFastPut = jsDynamicCast<JSFinalObject*>(target) && target->canPerformFastPutInlineExcludingProto() && target->isStructureExtensible();
+    unsigned argsCount = callFrame->argumentCount();
+
+    if (argsCount > 2 && argsCount < 5 && targetCanPerformFastPut) {
```

Comment:
Can you add a comment about why the selection of argCount between 2 and 5?  This is just a heuristic, right?  As in 1 means nothing to assign, at least 2 args before there is something to assign, and less than 5 limit the number of objects we'll need to scan this way and possibly fail out to the slow path anyway, correct?

Reply:
Right. 2 is `Object.assign(target, source)`. We do not need batching of multiple objects. If we increase this too much, we need to keep huge buffer, and we cut this with 5 threshold.